docs/self-managed/platform-deployment/helm-kubernetes/guides/connect-to-an-oidc-provider.md

```diff
@@ -0,0 +1,116 @@
+---
+id: connect-to-an-oidc-provider
+title: "Connect to an OpenID Connect provider"
+sidebar_label: "Connect to an OpenID Connect provider"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+To enable a smoother integration with your existing systems, Camunda supports connecting to an OpenID Connect
+authentication provider. In this guide we will walk you through the configuration required to connect Camunda to your
+authentication provider.
+
+:::note
+To connect to a Keycloak authentication provider, see [using an existing Keycloak](using-existing-keycloak.md).
+:::
+
+## Prerequisites
+
+- Information about your OpenID Connect provider's configuration, including:
+  - The issuer URL
+- Ability to create applications in your OpenID Connect provider
+- Ability to access the following information about the applications you have created in your OpenID Connect provider
+  - Client ID
+  - Client secrets
+  - Audience
+
+:::note
+The steps below are a general approach for the Camunda components, it is important you reference the [component-specific
+configuration](#component-specific-configuration) section to ensure the components are configured correctly.
+:::
+
+## Configuration
+
+<Tabs groupId="authPlatform" defaultValue="generic"
+values={[{label: 'Generic', value: 'generic' },{label: 'Microsoft Entra ID', value: 'microsoftEntraId' }]} >
+<TabItem value="generic">
+
+### Steps
+
+1. In your OpenID Connect provider, create an application for each of the components you want to connect. The expected
+   redirect URI of the component you are configuring an app for can be found
+   in [component-specific configuration](#component-specific-configuration).
+2. Make a note of the following values for each application you create:
+   - Client ID
+   - Client secret
+   - Audience
+3. Set the following environment variables for the component you are configuring an app for:
+
+```
+   CAMUNDA_IDENTITY_TYPE=GENERIC
+   CAMUNDA_IDENTITY_ISSUER=<URL_OF_ISSUER>
+   CAMUNDA_IDENTITY_ISSUER_BACKEND_URL=<URL_OF_ISSUER> // this is used for container to container communication
+   CAMUNDA_IDENTITY_CLIENT_ID=<Client ID from step 2>
+   CAMUNDA_IDENTITY_CLIENT_SECRET=<Client secret from step 2>
+   CAMUNDA_IDENTITY_AUDIENCE=<Audience from step 2>
+```
+
+### Additional considerations
+
+For authentication, the Camunda components use the scopes `email`, `openid`, and `offline_access`.
```

Comment:
@Ben-Sheppard It might be worth mentioning here that Web Modeler does not support authentication with a generic OIDC provider yet. This was not in scope of https://github.com/camunda/product-hub/issues/739.

Reply:
I'll take care of this in my technical review shortly 👍

Reply:
@wollefitz @Ben-Sheppard I was doing a final review of the C8SM and I saw that note about WM.
However it's not clear to me as a reader what should I do in that case when WM doesn't support generic OIDC :thinking:

Reply:
@aabouzaid this doesn't apply to the upcoming minor release anymore and will be removed once https://github.com/camunda/camunda-docs/pull/3041 is merged.

Reply:
@wollefitz Thanks for the confirmation, I will remove that note :ok_hand:

Reply:
@wollefitz Great, thanks for the confirmation ... Then is nothing to do from the Distro team since it will be removed in https://github.com/camunda/camunda-docs/pull/3041.